controllers/task.js

```diff
@@ -417,6 +418,126 @@ const getGroupByTasks = async (req, res, next) => {
 }
 exports.getGroupByTasks = getGroupByTasks;
 
+const exportDataToExcel = async (req, res, next) => {
+	let data = req.data;
+
+	let allowedTaskGroup = process.env.ALLOWED_GROUP_BY.split(',')
+
+	if (!allowedTaskGroup.includes(data.groupBy)) {
+		return res.status(400).send(sendResponse(400, `${data.groupBy} Group By Not Supported`, 'exportDataToExcel', null, req.data.signature))
+	} else {
+		data.aggregateGroupBy = `$${data.groupBy}`
+	}
+
+	data.aggregateGroupBy = CONSTANTS.GROUP_BY[data.groupBy]
+
+	let taskRes = await createPayloadAndGetGroupByTask(data)
+	let taskData = taskRes.data
+	// data.filteredData taskRes.data;
+	console.log(taskData.tasks)
+	if (taskRes.error) {
+		return res.status(500).send(sendResponse(500, '', 'exportDataToExcel', null, req.data.signature))
+	}
+
+	const workbook = new excelJS.Workbook();  // Create a new workbook
+	const worksheet = workbook.addWorksheet("My Users"); // New Worksheet
+	const path = "./";  // Path to download excel
+
+
+	// Column for data in excel. key must match data key
+	worksheet.columns = [
+		{ header: "S no.", key: "s_no", width: 20 },
+		{ header: "Project Name", key: "projectName", width: 20 },
+		{ header: "Section Name", key: "section", width: 20 },
+		{ header: "Completed Tasks", key: "completedTasks", width: 10 },
+		{ header: "Total Tasks", key: "totalTasks", width: 10 },
+		{ header: 'Task Title', key: 'taskTitle', width: 20 },
+		{ header: 'Task status', key: 'status', width: 20 },
+		{ header: 'Task description', key: 'description', width: 20 },
+		{ header: 'Task lead', key: 'lead', width: 20 },
+		{ header: 'Task CreatedBy', key: 'createdBy', width: 20 },
+		{ header: 'Task AssignedTo', key: 'assignedTo', width: 20 },
+		{ header: 'Task DueDate', key: 'dueDate', width: 20 },
+		{ header: 'Task Rated', key: 'isRated', width: 20 },
+		{ header: 'Task Rating', key: 'rating', width: 20 },
+		{ header: 'Task Created On', key: 'createdAt', width: 20 },
+
+
+	];
+
+	// Looping through User data
+	let counter = 1;
+	taskData.forEach((task) => {
+
+		let projectIdName = task._id.projectId;
+		let section = task._id.section;
+
+		task.projectName = projectIdName
+		task.section = section
+
+		task.tasks.forEach((tasks) => {
+			task.s_no = counter;
+			let taskId = tasks._id;
+			let taskTitle = tasks.title;
+			let description = tasks.description;
+			let status = tasks.status;
+			let lead = tasks.lead[0].name;
+
+			let assignedTo = ''
+			let assignedToUser = tasks.assignedTo?.name 
+			if (assignedToUser===null) {
+				assignedToUser = tasks.assignedTo;
+			}
+			assignedTo = assignedToUser;
+
+			let createdBy = ''
+			let createdByUser =  tasks.createdBy?.name 
```

Comment:
let createdByUser =  tasks.createdBy?.name 
			if (createdByUser===null) {
				createdByUser = tasks.createdBy;
			}
if name is coming everytime in createdBy as we must having name in creadtedBy object then there is no use of second if condition

Reply:
got it sir! I have removed if conditional